src/BenchmarkDotNet.Core/Portability/Cpu/MOSCpuInfoLoader.cs

```diff
@@ -0,0 +1,43 @@
+﻿#if !CORE
+using System.Collections.Generic;
+using System.Linq;
+using System.Management;
+using BenchmarkDotNet.Environments;
+
+namespace BenchmarkDotNet.Portability.Cpu
+{
+    public class MOSCpuInfoLoader : ICpuInfo
+    {
+        public MOSCpuInfoLoader()
```

Comment:
Could we hide the constructor and make it private?

In BenchmarkDotNet, we typically do something like this:
```cs
internal class MosCpuInfoLoader : ICpuInfo
{
    internal static readonly Lazy<ICpuInfo> Instance = new Lazy<ICpuInfo>(() => new MosCpuInfoLoader());
        
    private MosCpuInfoLoader()
    {
        // ...
```

It solves the following problems:
1. We try to ensure that we run such "heavy" logic only once.
2. RuntimeInformation logic shouldn't allocate additional memory after warmup (it can affect MemoryDiagnoser)
3. With the private constructor and internal singleton, a user can't accidentally invoke it and get an exception on a wrong OS.
4. Users don't need *public* OS-specific classes; they should get all information via "recommended" API. All magic should be hidden inside.
5. (A note): we typically use `Mos` notation instead of `MOS` for three-letter abbreviations.

P.S. Please, make the changes for other `ICpuInfo` implementations as well.

Reply:
@AndreyAkinshin How do you think MosCpuInfoLoader has to implement ICpuInfo or it has to have method Load() which returns CpuInfo (from the next comment)?

Reply:
@morgan-kn, I think, we can write something like this (it should be consistent):

```cs
internal static class MosCpuInfo
{
    internal static Lazy<ICpuInfo> Instance = new Lazy<ICpuInfo>(Load);

    private static ICpuInfo Load()
    {
        var processorModelNames = new HashSet<string>();
        uint physicalCoreCount = 0;
        uint logicalCoreCount = 0;
        int processorsCount = 0;

        var mosProcessor = new ManagementObjectSearcher("SELECT * FROM Win32_Processor");
        foreach (var moProcessor in mosProcessor.Get().Cast<ManagementObject>())
        {
            var name = moProcessor[WmicCpuInfoKeyNames.Name]?.ToString();
            if (!string.IsNullOrEmpty(name))
            {
                processorModelNames.Add(name);
                processorsCount++;
                physicalCoreCount += (uint) moProcessor[WmicCpuInfoKeyNames.NumberOfCores];
                logicalCoreCount += (uint) moProcessor[WmicCpuInfoKeyNames.NumberOfLogicalProcessors];
            }
        }

        return new CpuInfo(
            processorModelNames.Count > 0 ? string.Join(", ", processorModelNames) : null,
            processorsCount > 0 ? processorsCount : (int?) null,
            physicalCoreCount > 0 ? (int?) physicalCoreCount : null,
            logicalCoreCount > 0 ? (int?) logicalCoreCount : null);
    }
}
```

Reply:
@AndreyAkinshin  Yes, I thought about it, but is that ok that field Instance in MosCpuInfo class will have another type but not MosCpuInfo?

Reply:
@morgan-kn, ok, what do you think about such naming:
```
internal static class MosCpuInfoProvider
{
    internal static Lazy<ICpuInfo> MosCpuInfo = new Lazy<ICpuInfo>(Load);
```

Reply:
@AndreyAkinshin ok, sounds good, thanks